firebase-vscode/webviews/EmulatorPanel.tsx

```diff
@@ -0,0 +1,249 @@
+    console.log(`notifyEmulatorImportFolder received in sidebar: ${folder}`);
+    emulatorUiSelections.importStateFolderPath = folder;
+    setEmulatorUiSelectionsAndSaveToWorkspace({ ...emulatorUiSelections }); // rerender clone
+  });
+
+  function launchEmulators() {
+    if (!emulatorUiSelections.projectId) {
+      broker.send("showMessage", {
+        msg: "Missing project ID",
+        options: {
+          modal: true,
+          detail: `Please specify a project ID before starting the emulator suite.`,
+        },
+      });
+      return;
+    }
+    if (!firebaseJson) {
+      // TODO(christhompson): Consider using a default config in the case that
+      // firebase.json doesnt exist.
+      broker.send("showMessage", {
+        msg: "Missing firebase.json",
+        options: {
+          modal: true,
+          detail: `Unable to find firebase.json file.`,
+        },
+      });
+      return;
+    }
+    setShowEmulatorProgressIndicator(true);
+    broker.send("launchEmulators", {
+      firebaseJson,
+      emulatorUiSelections,
+    });
+  }
+
+  function stopEmulators() {
+    setShowEmulatorProgressIndicator(true);
+    broker.send("stopEmulators");
+  }
+
+  /**
+   * Called when import folder changes.
+   */
+  function selectedImportFolder(event: any) {
+    event.preventDefault();
+    broker.send("selectEmulatorImportFolder");
+  }
+
+  function toggleExportOnExit() {
+    console.log("toggle export on exit");
+    const selections: EmulatorUiSelections = emulatorUiSelections;
+    selections.exportStateOnExit = !selections.exportStateOnExit;
+    setEmulatorUiSelectionsAndSaveToWorkspace(selections);
+  }
+
+  function projectIdChanged(event: React.ChangeEvent<HTMLInputElement>) {
+    console.log("projectIdChanged: " + event.target.value);
+    const selections: EmulatorUiSelections = emulatorUiSelections;
+    selections.projectId = event.target.value;
+    setEmulatorUiSelectionsAndSaveToWorkspace(selections);
+  }
+
+  function emulatorModeChanged(event: React.ChangeEvent<HTMLSelectElement>) {
+    console.log("emulatorModeChanged: " + event.target.value);
+    const selections: EmulatorUiSelections = emulatorUiSelections;
+    selections.mode = event.target.value as typeof emulatorUiSelections.mode;
+    setEmulatorUiSelectionsAndSaveToWorkspace(selections);
+  }
+
+  function clearImportFolder() {
+    console.log(`clearImportFolder`);
+    emulatorUiSelections.importStateFolderPath = "";
+    setEmulatorUiSelectionsAndSaveToWorkspace({ ...emulatorUiSelections });
+  }
+
+  // Make it pretty for the screen. Filter out the logging emulator since it's
+  // an implementation detail.
+  // TODO(christhompson): Add more info and sort this.
+  function formatEmulatorRunningInfo(emulatorInfos: EmulatorInfo[]): string {
+    return emulatorInfos
+      .map((info) => info.name)
+      .filter((name) => name !== "logging")
+      .join("<br/>");
+  }
+
+  return (
+    <PanelSection>
```

Comment:
Can you pass it a `title` prop? Then it'll have a nice header bar and also be collapsible.

Reply:
Done